### PR TITLE
[Telink] Fix DFU target build & add to CI

### PR DESCRIPTION
**Pull request by s07641069**
**Change overview**

- Add DFU target build to CI
- Fix DFU target build
- Cleanup merging of binaries

### CODE REVIEW

File: src/platform/telink/OTAUtil.cpp [hunk 1]
```diff
@@ -20,7 +20,6 @@
 #include <app/clusters/ota-requestor/DefaultOTARequestor.h>
 #include <app/clusters/ota-requestor/DefaultOTARequestorDriver.h>
 #include <app/clusters/ota-requestor/DefaultOTARequestorStorage.h>
-#include <app/server/Server.h>
```
Comment (andy31415) [inline src/platform/telink/OTAUtil.cpp:23]: How about the include below for MCUBOOT?

File: .github/workflows/examples-telink.yaml [hunk 2]
```diff
@@ -139,10 +139,10 @@ jobs:
             - name: Build example Telink (B92) Light Switch App with OTA, Shell, Factory Data
               run: |
                   ./scripts/run_in_build_env.sh \
-                    "./scripts/build/build_examples.py --target 'telink-tlsr9528a-light-switch-ota-shell-factory-data' build"
+                    "./scripts/build/build_examples.py --target 'telink-tlsr9528a-light-switch-shell-factory-data' build"
                   .environment/pigweed-venv/bin/python3 scripts/tools/memory/gh_sizes.py \
-                    telink tlsr9528a light-switch-app-ota-shell-factory-data \
-                    out/telink-tlsr9528a-light-switch-ota-shell-factory-data/zephyr/zephyr.elf \
+                    telink tlsr9528a light-switch-app-shell-factory-data \
```
Comment (andy31415) [inline .github/workflows/examples-telink.yaml:144]: Why is this to be removed?
Is this unsupported?

**Review by andy31415 — COMMENTED** [review]
@s07641069 could you add more details in the summary? what is a "build conflict"?

This change is not obvious to understand

**Comment by s07641069** [comment]
> add more details in the summary? what is a "build conflict"?

"#include <app/server/Server.h>" needed to build config with disabled CONFIG_CHIP_OTA_REQUESTOR and enabled CONFIG_BOOTLOADER_MCUBOOT.
But this config build is not included in GitHub CI.

Lets use your fix of CI builds https://github.com/project-chip/connectedhomeip/pull/29750

I will add this config build to GitHub CI after resolve of build conflicts.

**Review by andy31415 — APPROVED** [review]

**Pull request merged**